cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/SparkPipelineRunner.java

```diff
@@ -546,7 +546,8 @@ protected SparkCollection<Object> handleJoin(Map<String, SparkCollection<Object>
       Map<String, Schema> inputSchemas = new HashMap<>();
       for (String inputStageName : pipelinePhase.getStageInputs(stageName)) {
         StageSpec inputStageSpec = pipelinePhase.getStage(inputStageName);
-        inputSchemas.put(inputStageName, inputStageSpec.getOutputSchema());
+        inputSchemas.put(inputStageName,
+                         inputStageSpec.getOutputPorts().get(stageName).getSchema());
```

Comment:
will the output ports map always contain the stage name as a key, even when the input schema actually is null? 

I'm thinking it would be a little safer to check for null before calling `.getSchema()`

Reply:
If we got a null when get(stageName), we should have a exception. Is that correct?